homeassistant/components/prosegur/__init__.py

```diff
@@ -0,0 +1,50 @@
+"""The Prosegur Alarm integration."""
+import logging
+
+from pyprosegur.auth import Auth
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client
+
+from .const import CONF_COUNTRY, DOMAIN
+
+PLATFORMS = ["alarm_control_panel"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Prosegur Alarm from a config entry."""
+    try:
+        session = aiohttp_client.async_get_clientsession(hass)
+        hass.data.setdefault(DOMAIN, {})
+        hass.data[DOMAIN][entry.entry_id] = Auth(
+            session,
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+            entry.data[CONF_COUNTRY],
+        )
+        await hass.data[DOMAIN][entry.entry_id].login()
+
+    except ConnectionRefusedError as error:
+        _LOGGER.error("Configured credential are invalid, please reconfigure")
+        raise ConfigEntryAuthFailed from error
```

Comment:
Not for now, but for a future PR, you could implement authentication here instead

Reply:
Any existing integration with similar pattern I could look at ?

Reply:
Any integration that implemenets `async_step_reauth` in its config flow.

Reply:
I implemented `async_step_reauth` following `verisure`

But I'm not sure how does it get triggered

Reply:
Verisure uses a DataUpdateCoordinator, which will trigger that in case the first update fails on that. 

You can trigger a config flow at this point in the code, e.g. using

Reply:
Thanks!

Reply:
No problem